TestingLowerBounds/Divergences/StatInfo.lean

```diff
@@ -145,6 +146,252 @@ lemma toReal_statInfo_eq_min_sub_integral (μ ν : Measure 𝒳) [IsFiniteMeasur
+          (Measure.measurable_rnDeriv μ ν).aemeasurable] with x hx1 hx2 hx3
+        intro hxs
+        nth_rw 1 [← Measure.singularPart_add_rnDeriv μ ν]
+        simp_rw [hx2, Pi.add_apply, hx3, hx1 hxs, mul_zero, ENNReal.zero_toReal, sub_zero, add_zero]
+      · apply setIntegral_congr_ae hs.compl
+        filter_upwards [Measure.rnDeriv_restrict μ (twoHypKernel μ ν ∘ₘ π) hs.compl,
+          Measure.rnDeriv_mul_rnDeriv
+          (Measure.absolutelyContinuous_restrict_compl_singularPartSet μ ν)] with x hx1 hx2 hxs
+        rw [max_mul_of_nonneg _ _ ENNReal.toReal_nonneg, zero_mul, sub_mul]
+        rw [Set.indicator_of_mem hxs] at hx1
+        simp_rw [ENNReal.toReal_mul, mul_assoc, ← hx1, ← hx2, Pi.mul_apply, ENNReal.toReal_mul]
+    _ = ∫ x, max 0 (π {false} * (∂(μ.singularPart ν)/∂twoHypKernel μ ν ∘ₘ π) x).toReal
+          ∂twoHypKernel μ ν ∘ₘ π
+        + ∫ x in sᶜ, (max 0 ((π {false}
+          * (∂μ.restrict (μ.singularPartSet ν)ᶜ/∂ν) x).toReal - (π {true}).toReal)) ∂ν := by
+      congr 1
+      · nth_rw 2 [← integral_add_compl hs]
+        swap
+        · simp_rw [ENNReal.toReal_mul]
+          exact ((integrable_zero _ _ _).sup (Measure.integrable_toReal_rnDeriv.const_mul _))
+        refine self_eq_add_right.mpr <| setIntegral_eq_zero_of_ae_eq_zero ?_
+        filter_upwards [Measure.rnDeriv_restrict μ _ hs] with x hx
+        intro hxs
+        rw [← Measure.restrict_singularPartSet_eq_singularPart, hx, indicator_of_not_mem hxs,
+          mul_zero, ENNReal.zero_toReal, max_self]
+      · simp_rw [mul_comm _ ((∂ν/∂_ ∘ₘ _) _).toReal, ← smul_eq_mul]
+        rw [setIntegral_rnDeriv_smul hνac hs.compl]
+    _ = ∫ x, (π {false} * (∂(μ.singularPart ν)/∂twoHypKernel μ ν ∘ₘ π) x).toReal
+          ∂twoHypKernel μ ν ∘ₘ π
+        + ∫ x in sᶜ, (max 0 ((π {false} * (∂μ/∂ν) x).toReal - (π {true}).toReal)) ∂ν := by
+      simp_rw [max_eq_right ENNReal.toReal_nonneg]
+      congr 1
+      apply setIntegral_congr_ae hs.compl
+      filter_upwards [Measure.rnDeriv_restrict μ ν hs.compl] with x hx1 hxs
+      rw [hx1, indicator_of_mem hxs]
+    _ = ∫ x, (max 0 ((π {false} * (∂μ/∂ν) x).toReal - (π {true}).toReal)) ∂ν
+        + (π {false} * (μ.singularPart ν) univ).toReal := by
+      simp_rw [ENNReal.toReal_mul, add_comm (∫ _, _ ∂_ ∘ₘ _)]
+      rw [integral_mul_left, Measure.integral_toReal_rnDeriv
+        ((Measure.singularPart_le μ ν).absolutelyContinuous.trans hμac)]
+      nth_rw 2 [← integral_add_compl hs]
+      swap
+      · exact (integrable_zero _ _ _).sup
+          ((Measure.integrable_toReal_rnDeriv.const_mul _).sub (integrable_const _))
+      rw [setIntegral_zero_measure _ (Measure.measure_singularPartSet μ ν), zero_add]
+
+
+lemma toReal_statInfo_eq_integral_max_of_gt {μ ν : Measure 𝒳} [IsFiniteMeasure μ] [IsFiniteMeasure ν]
+    {π : Measure Bool} [IsFiniteMeasure π] (h : π {true} * ν univ < π {false} * μ univ) :
+    (statInfo μ ν π).toReal
+      = ∫ x, max 0 ((π {true}).toReal - (π {false} * (∂μ/∂ν) x).toReal) ∂ν := by
```

Comment:
Is it possible to get a shorter proof of that one with `statInfo_symm` and `toReal_statInfo_eq_integral_max_of_le`?